pkg/bpf/collection.go

```diff
@@ -287,6 +296,31 @@ func renameMaps(coll *ebpf.CollectionSpec, renames map[string]string) error {
 	return nil
 }
 
+func serializeConstants(logger *slog.Logger, opts *CollectionOptions) {
```

Comment:
I would return errors instead of logging and quietly continuing. CI jobs fail on `level=error`, but in a user's env, this likely won't raise alarm bells. A failure to write to the statedir is an indication that something else is afoul; we shouldn't happily continue in that case.